addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/DiceMatcher.java

```diff
@@ -42,43 +42,36 @@ private DiceMatcher() {}
      * License :  https://creativecommons.org/licenses/by-sa/3.0/
      * Author : Jelle Fresen
      * Changes : Fixed indexing to prevent out of array access
+     *           Address various issues identified by static analysis
      * Released under CC-BY-SA.
      */
 
     public static int getMatchPercentage(String a, String b) {
 
+        // Verifying the input:
+        if (a == null || b == null) return 0;
+
         String s = a.replaceAll("\\s+", " ");
         String t = b.replaceAll("\\s+", " ");
 
-        // Verifying the input:
-        if (s == null || t == null) return 0;
         // Quick check to catch identical objects:
-        if (s == t) return 1;
+        if (s.equals(t)) return 100;
```

Comment:
Worth adding a test. The previous check can be kept (which can be tested too).

Reply:
Also, the comment is no longer accurate.

Reply:
You lost me, why would we put the `==` back, that was the SAST issue being addressed.

Reply:
`==` checks identity not equality (i.e. if the same string is being checked).

True, then just needs for the identity one.

Reply:
Although equals will already do that.... guess it's just left the comment.

Reply:
This is no longer a quick check (not as the identity).